src/webhook/mutation/pod_mutator.go

```diff
@@ -211,34 +213,44 @@ func (m *podMutator) Handle(ctx context.Context, req admission.Request) admissio
 		return *nsResponse
 	}
 
+	m.currentPodNS = ns.Name
+	defer func() {
+		m.currentPodNS = ""
+	}()
```

Comment:
this defer function seems unnecessary

Reply:
@meik99 I was simply following the pattern from https://github.com/Dynatrace/dynatrace-operator/blob/master/src/webhook/mutation/pod_mutator.go#L200-L202

I agree that it's probably unnecessary if a new podMutator object is created (and likely garbage collected afterward) every time the `Handler()` function is called and the new struct instance of course has zero values for these fields.

It did make me stop to wonder why `m.currentPodName = ""` was being deferred in the original code (does this not get a new podMutator on every new `Handler()` call?), and I took the precaution against leaking old/outdated pods that were previously injected. It was just a safer bet to follow the same pattern, IMHO.

If you disagree, I'm happy to remove it, but we should probably also remove the above linked `defer func()` for nullifying `m.currentPodName` also.

Please advise.

Reply:
please remove both defers, they are weird

Reply:
@meik99 @chrismuellner I've removed all deferred functions where podMutator fields are nullified.